src/lava/lib/optimization/solvers/bayesian/solver.py

```diff
@@ -0,0 +1,293 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: LGPL-2.1-or-later
+# See: https://spdx.org/licenses/
+
+import numpy as np
+import os
+
+from schema import And, Schema, SchemaError
+
+from lava.magma.core.process.process import AbstractProcess
+from lava.magma.core.run_conditions import RunSteps
+from lava.magma.core.run_configs import Loihi1SimCfg
+
+from lava.lib.optimization.solvers.bayesian.models import BayesianOptimizer
+
+
+class BayesianSolver:
+    """
+    The BayesianSolver is a class-based interface to abstract the details
+    of initializing the BayesianOptimizer process and connecting it with
+    the user's specific black-box function
+    """
+    def __init__(self, acq_func_config: dict, acq_opt_config: dict,
```

Comment:
Stylistically this constructor might be improved by pushing the validation Schema objects up to static class members or blocking their creation followed by the actual validation... It's a lot of repetition/boilerplate without actually doing much.